src/mpd.c

```diff
@@ -3725,23 +3737,39 @@ mpd_command_toggleoutput(struct evbuffer *evbuf, int argc, char **argv, char **e
  * Example output:
  *   outputid: 0
  *   outputname: Computer
+ *   plugin: alsa
  *   outputenabled: 1
  *   outputvolume: 50
+ * https://mpd.readthedocs.io/en/latest/protocol.html#audio-output-devices
  */
 static void
 speaker_enum_cb(struct player_speaker_info *spk, void *arg)
 {
-  struct evbuffer *evbuf;
+  struct output_outputs_param *param = arg;
+  struct evbuffer *evbuf = param->buf;
+  char plugin[sizeof(spk->output_type) + 1];
```

Comment:
You don't need +1 here. sizeof() returns a size that has room for the zero terminator.

Reply:
output_type is a char[50] so sizeof will return 50 here
plugin is unconditionally appended a '\0' to
but you can argue I don't check I actually overrun output_type and thus it must have a terminating '\0', so 50 is sufficient.

Reply:
It's ok to assume here that output_type is a valid C string. It's the job of the other parts of OwnTone that set the value of this string to assure that.